app/jobs/itms/sync_discrepancy_job.rb

```diff
@@ -0,0 +1,87 @@
+require 'harvester_utils/downloader'
+
+class Itms::SyncDiscrepancyJob < ItmsJob
+  def perform(itms_id, downloader: HarvesterUtils::Downloader)
+    response = downloader.get("https://opendata.itms2014.sk/v2/nezrovnalost/#{itms_id}")
+    json = JSON.parse(response.body)
+
+    ActiveRecord::Base.transaction do
+      d = Itms::Discrepancy.find_or_initialize_by(itms_id: itms_id)
+      d.itms_href = json['href']
+      d.itms_created_at = json['createdAt']
+      d.itms_updated_at = json['updatedAt']
+      d.save!
```

Comment:
@jsuchal Tymto (skory zapis nekompletneho zaznamu v transakcii) by som sa vyhol deadlockom (ked Objekt A potrebuje Objekt B a Objekt B potrebuje Objekt A).

Takto mozem vytvorit vztahy v self-referential join tabulkach (napr. `nezrovnalost.suvisiace_nezrovnalosti`) predtym nez su tie objekty kompletne vyskladane.

Reply:
Asi uplne nerozumiem ako to riesi ten deadlock, kedze transakciu nevidiet pokym nie je commit.

Reply:
Hej, riesi to zrejme len deadlock v ramci jednej transakcie. Na paralelne transakcie budem musiet vymysliet nieco lepsie.